spec/03-plugins/10-basic-auth/00-migrations/003_200_to_210_spec.lua

```diff
@@ -0,0 +1,116 @@
+      assert.equals(default_ws_id, a.ws_id)
+
+      -- create entities without specifying default ws_id (simulate old node)
+      local a = h.pg_insert(cn, "basicauth_credentials", { id = utils.uuid() })
+      assert.equals(default_ws_id, a.ws_id)
+
+      -- create specifying default ws_id.(simulate new node)
+      local a = h.pg_insert(cn, "basicauth_credentials", { id = utils.uuid(), ws_id = default_ws_id })
+      assert.equals(default_ws_id, a.ws_id)
+
+      -- kong migrations finish
+      assert(helpers.run_teardown_migration(db, "basic-auth",
+                                            "kong.plugins.basic-auth.migrations",
+                                            "003_200_to_210"))
+
+      -- create specifying default ws_id.(simulate new node)
+      local a = h.pg_insert(cn, "basicauth_credentials", { id = utils.uuid(), ws_id = default_ws_id })
+      assert.equals(default_ws_id, a.ws_id)
+    end)
+  end)
+
+  describe("#cassandra", function()
+    before_each(function()
+      _, db = helpers.get_db_utils("cassandra", nil, nil, {
+        stop_namespace = "kong.plugins.basic-auth.migrations",
+        stop_migration = "003_200_to_210",
+      })
+    end)
+
+    it("adds and sets ws_id", function()
+      local cn = db.connector
+      h.assert_not_c_has_fkey(cn, "basicauth-credentials", "ws_id")
+      -- kong migrations up
+      assert(helpers.run_up_migration(db, "basicauth-credentials",
+                                      "kong.plugins.basic-auth.migrations",
+                                      "003_200_to_210"))
+
+      h.c_insert(cn, "basicauth_credentials", { id = utils.uuid() })
+
+      -- MIGRATING
+      h.assert_c_has_fkey(cn, "basicauth_credentials", "ws_id")
+
+      -- check default workspace exists and get its id
+      local res = assert(cn:query("SELECT * FROM workspaces"))
+      assert.equals(1, #res)
+      assert.equals("default", res[1].name)
+      assert.truthy(utils.is_valid_uuid(res[1].id))
+      local default_ws_id = res[1].id
+
+      -- entities created by the old node don't get the default id in C*
+      -- (this is handled in the DAO)
+      local a = assert(cn:query("SELECT * FROM basicauth_credentials"))[1]
+      assert.is_nil(a.ws_id)
+
+      -- create entities without specifying default ws_id (simulate old node)
+      local a = h.c_insert(cn, "basicauth_credentials", { id = utils.uuid() })
+      assert.is_nil(a.ws_id)
+
+      -- create specifying default ws_id.(simulate new node)
+      local a = h.c_insert(cn, "basicauth_credentials", { id = utils.uuid(), ws_id = default_ws_id })
+      assert.equals(default_ws_id, a.ws_id)
+
+      -- kong migrations finish
+      assert(helpers.run_teardown_migration(db, "basic-auth",
+                                            "kong.plugins.basic-auth.migrations",
+                                            "003_200_to_210"))
+
+      -- create specifying default ws_id.(simulate new node)
```

Comment:
Let's add a test that entities created by old nodes receive a ws_id after teardown (applies to other plugins tests as well)